src/Raven.Server/Documents/ClusterTransactionWaiter.cs

```diff
@@ -34,7 +34,7 @@ public void SetResult(string id, long index, object result)
             Database.RachisLogIndexNotifications.NotifyListenersAbout(index, null);
             if (_results.TryGetValue(id, out var task))
             {
-                task.SetResult(result);
+                task.TrySetResult(result);
```

Comment:
Are we expecting this to be called multiple times? We might be hiding another issue here.

Reply:
It should't be set multiple times